src/program/calibration.rs

```diff
@@ -91,15 +120,90 @@ impl CalibrationSet {
                                     }
                                 }
                             }
+
+                            instruction.apply_to_expressions(|expr| {
+                                let previous = std::mem::replace(expr, Expression::PiConstant);
+                                *expr = previous.substitute_variables(&variable_expansions);
+                            })
```

Comment:
I tried that first, and for reasons I can't recall now, couldn't get it to work or maybe it was just clumsy. I do recall a nuance about how ownership works within closures because the compiler has to consider the case in which the closure panics. I had a comment to that effect on this PR but it was still "pending" until just now - if you or @nilslice find a more ergonomic way I'd be happy to consider that instead. Otherwise, we might want to change the ownership-taking methods on `Instruction` to instead take mutable references.

Reply:
I'd say worth adding a comment for now clarifying why this does what it does and we can come back and improve ergonomics in the future (unless @nilslice has time to look at it before this needs to merge).

I assume there's some sort of lifetime issue with this?

```suggestion
                            instruction.apply_to_expressions(|expr| {
                                *expr = expr.substitute_variables(&variable_expansions);
                            })
```

Reply:
If general consensus is OK with this as-is, I'd prefer merging it without doing a deep-dive while I'm on another track. After `pyquil` benchmarks are added I'll be back into this repo heads-down and can take a closer look.